stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -299,6 +309,81 @@ public struct TaskGroup<ChildTaskResult: Sendable> {
   ///   - operation: The operation to execute as part of the task group.
   /// - Returns: `true` if the child task was added to the group;
   ///   otherwise `false`.
+  @_alwaysEmitIntoClient
+  @_allowFeatureSuppression(IsolatedAny)
+  public mutating func addTaskUnlessCancelled(
+    priority: TaskPriority? = nil,
+    operation: __owned @Sendable @escaping @isolated(any) () async -> ChildTaskResult
+  ) -> Bool {
+#if compiler(>=5.5) && $BuiltinCreateAsyncTaskInGroup
+    let canAdd = _taskGroupAddPendingTask(group: _group, unconditionally: false)
+
+    guard canAdd else {
+      // the group is cancelled and is not accepting any new work
+      return false
+    }
+#if SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
+    let flags = taskCreateFlags(
+      priority: priority, isChildTask: true, copyTaskLocals: false,
+      inheritContext: false, enqueueJob: false,
+      addPendingGroupTaskUnconditionally: false,
+      isDiscardingTask: false)
+#else
+    let flags = taskCreateFlags(
+      priority: priority, isChildTask: true, copyTaskLocals: false,
+      inheritContext: false, enqueueJob: true,
```

Comment:
Just an idea: I wonder if it's time to make some

```swift
@inline(__always)
@inlinable
var SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY: Bool { 
#if SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
  true
#else 
  false
#endif 
}
```

because this entire #ifelse is really just to change the `enqueueJob` depending on the env...?

(Trying to get less of those #ifelse in this file, it's getting somewhat intense...)

Reply:
Yeah, think there's a lot of redundancy we could eliminate with the flags.

Reply:
Yeap some soon™ cleanups then... No problem in this PR though, land as is :)